doc/advanced/content/executor_design.rst

```diff
@@ -346,74 +342,73 @@ Best Fit Executor
-all the common cod to be shared among all the executors. The base executors has a CRTP based design which
-is used to access the properties of the derived executor. It also allowed simplification in many areas
-such as being able to reference any derived executor. Basic properties of all executors like
+The idea was to have a single base executor from which all executors would derive from. This would have allowed
+all the common code to be shared among all the executors. The base executor had a CRTP based design that
+was used to access the properties of the derived executor. It also allowed simplification in many areas, such as being able to reference any derived executor. Basic properties of all executors like
 copy constructors, overloaded equality operators.
-The CRTP mechanism had some restrictions in the sense that there was still a need for templates and
-their were concerns regarding explicitly passing all the properties to the base class using CRTP as
+The CRTP mechanism had some restrictions in the sense that there was still a need for templates, and
+there were concerns regarding explicitly passing all the properties to the base class using CRTP as
 it was felt to be unnecessary. Having inheritance also introduces runtime polymorphism as the derived executor
-would override some of the base executors method, and the call to the overridden function is resolved at run time.
-This was opposition to one of the design consideration i.e. have everything compile time and avoid any overhead.
-The base executor didn't add a lot besides allowing code sharing of a few common functionality and thus
+would override some of the base executors' method, and the call to the overridden function is resolved at run time.
+This is opposed to one of the design consideration i.e., have everything compile-time and avoid any overhead.
+The base executor didn't add a lot besides allowing code sharing of a few common functionalities and thus
 it was discarded.
 
 * Property Inheritance
 
 In this design, the executor inherits from the properties it supports (which are defined as template parameters
 for the executor class). With this design, executors would not need to provide the `require` or `prefer`
 and `query` member functions inside the executor for each property it wants to allow association with and
-support customizing. This design was not fully explored, so the potential drawbacks are not know completely.
-A snippet is available `here <https://godbolt.org/z/zhKM6e>`_. It was decided to not go with this design forward
-due to the potential complications, minimal support of properties for executors in PCL and deviation
+support customizing. This design was not fully explored, so the potential drawbacks are not known completely.
+A snippet is available `here <https://godbolt.org/z/zhKM6e>`_. It was decided not to go forward with this design
+due to the potential complications, minimal support of properties for executors in PCL, and deviation
```

Comment:
potential complications doesn't really add any info to this sentence

Reply:
Removed and reworded